code/modules/admin/IsBanned.dm

```diff
@@ -1,32 +1,51 @@
 //Blocks an attempt to connect before even creating our client datum thing.
-world/IsBanned(key,address,computer_id)
-	if(ckey(key) in admin_datums)
+/world/IsBanned(key, address, computer_id, type)
+	var/static/key_cache = list()
+	if(type == "world")
+		return ..()
+
+	if(key_cache[key] >= REALTIMEOFDAY)
+		return list("reason"="concurrent connection attempts", "desc"="You are attempting to connect too fast. Try again.")
+	key_cache[key] = REALTIMEOFDAY + 10 //This proc shouldn't be runtiming. But if it does, then the expiry time will cover it to ensure genuine connection attempts don't get trapped in limbo.
+
+	var/ckeytext = ckey(key)
+
+	if(ckeytext in admin_datums)
+		key_cache[key] = 0
 		return ..()
 
 	//Guest Checking
 	if(!config.guests_allowed && IsGuestKey(key))
 		log_access("Failed Login: [key] - Guests not allowed")
 		message_admins("<span class='notice'>Failed Login: [key] - Guests not allowed</span>")
+		key_cache[key] = 0
```

Comment:
Why null this for guests?

Reply:
Mainly so that guest ckeys don't get stuck in limbo when getting bounced off a guest wall. Less of an issue since key_cache was given expiry times in this implementation, but still worth keeping into account of regardless.